src/expand_tilde.c

```diff
@@ -0,0 +1,140 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   expand_tilde.c                                     :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/07/13 22:15:14 by gwolf             #+#    #+#             */
+/*   Updated: 2023/07/14 19:38:20 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file expand_tilde.c
+ * @brief Functions to handle expansion of tilde.
+ */
+#include	"expander.h"
+
+/**
+ * @brief Find the corresponding replacement for token.
+ *
+ * Depending on the token find env var.
+ * ~: $HOME.
+ * ~+: $PWD.
+ * ~-: $OLDPWD.
+ * If not found increase update position to account for '~' and
+ * return ERR_NOT_FOUND.
+ * If found ft_strdup() the value.
+ * Calc replace.len.
+ *
+ * @param token Contains searched for token.
+ * @param symtab Environment.
+ * @param replace Where to save replace string.
+ * @param pos Current position.
+ * @return t_error SUCCESS, ERR_NOT_FOUND, ERR_MALLOC
+ */
+t_error	ft_get_tilde_replace(t_tok token, t_hashtable *symtab, t_tok *replace, size_t *pos)
+{
+	char		*target;
+	t_env_var	*env_var;
+
+	if (!ft_strcmp(token.str, "~"))
```

Comment:
I think it's neater to read, since it doesn't involve another function call inside the function. 
Also it doesn't need ft_strlen() since it's running as long as the string isn't terminated. That means for every call of strcmp() we save a strlen() which means a speed increase of 100%!

Reply:
I gonna delete the strcmp 🤣

Reply:
😂

Reply:
ft_strcmp has been completely eradicated 💇